_posts/2019-10-27-on-code-reviews.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: post
+permalink: on-code-reviews
+title: 'On Code Reviews'
+tags: [programming]
+---
+
+I wanted to share some insights into how I review code, learning from others,
+helping others, and getting a better overview of the code base which ultimately
+allows me to write better[^1] code. This post is inspired by a recent
+[tweet](https://twitter.com/kristoferbaxter/status/1186290473322741760), a
+similar [blog post](https://techbeacon.com/app-dev-testing/10-commandments-navigating-code-reviews)
+(by [Angie Jones](https://twitter.com/techgirl1908)) and a [talk](https://www.youtube.com/watch?v=Ea8EiIPZvh0)
+on the topic (by [April Wensel](https://twitter.com/aprilwensel)) from 2018's
+[try! Swift NYC](https://www.tryswift.co) (which you should totally attend if
+you get the chance).
+
+- **I point out the good things.** This goes for anything really; not just code
+reviews. If there's something you like or learned, let the author know. I
+believe this will make a big difference on how a review is perceived, and is a
+win-win all around.<br />
+This is what I believe to be the **single most important part of a review**.
+- **I assume best intent.** I've written questionable code before, or have
+overseen mistakes. That happens. Always be kind and approach a review
+positively. 
+- **I read the pull request's description _and_ ticket.** Yes, also the ticket.
+This makes the time of everyone involved a lot more valuable. Seldom does this
+not help me getting a better understanding of the code. And understanding is
+key. If something in the description or ticket is unclear to me, I try to reach
+out the author before continuing a review.
+- **I provide context.** For instance, if I come across something that might
+better be represented by a function rather than a computed property, I explain
+why I would suggest that in the given case.
+- **I provide references.** Whether it's a blog post, piece of code, or
+documentation I reference, I link it. This allows the author to dig deeper into
+the topic if they want, and may be helpful for those who read the comment(s) in
+the future.
+- **I take the time to try and understand the code.** When I feel like I'm
+having a hard time grasping some of the decisions that have been made, or the
+code itself, I often feel asking about it in writing is difficult. If this
+happens, I either take a step away from the computer and try again later, or ask
+to go over it in person (face-to-face or in a video call, _not_ a phone call.
+Being able to see each other aids understanding).<br />
+An exception to this would be a pull request I feel is introducing too many
+concepts at once, in which case I'll either ask if we could break up the pull
+request in more bite-sized pieces, offering to help do so in person if need be.
```

Comment:
I personally find it very difficult to break up a PR that has gotten out of hand, maybe you could provide some tips here

Reply:
I'd love to go follow up on that some time, but won't do so here. It's difficult, and certainly a case of "prevention is better than cure". It depends on how an author breaks up commits... so many factors.